### ISSUE TITLE
Linux AppImage fails to start

### ISSUE THREAD

**Issue by mrbumpy409** [audacity/audacity#4629]
### Bug description

I have several Ubuntu-based Linux installs, and the current 3.3.1 AppImage refuses to start in any of them. This includes the following operating systems, all up-to-date:

- KDE neon 5.27 (my main desktop system)
- KDE neon 5.27 (VirtualBox)
- Kubuntu 22.04 (VirtualBox)
- Ubuntu 22.04 (VirtualBox)
- Ubuntu Studio 23.04 (VirtualBox)
- Xubuntu 22.04 (VirtualBox)

I have tried booting into both X11 and Wayland with the same result. Here is the terminal output:

```
/lib/x86_64-linux-gnu/libatk-1.0.so.0
/lib/x86_64-linux-gnu/libatk-bridge-2.0.so.0
/lib/x86_64-linux-gnu/libcairo-gobject.so.2
/lib/x86_64-linux-gnu/libcairo.so.2
/lib/x86_64-linux-gnu/libgio-2.0.so.0
/lib/x86_64-linux-gnu/libglib-2.0.so.0
/lib/x86_64-linux-gnu/libgmodule-2.0.so.0
/lib/x86_64-linux-gnu/libgobject-2.0.so.0
/lib/x86_64-linux-gnu/libgthread-2.0.so.0
/lib/x86_64-linux-gnu/libjack.so.0
findlib: libportaudio.so: cannot open shared object file: No such file or directory
/home/chris/Downloads/audacity-linux-3.3.1-x64.AppImage: Using fallback for library 'libportaudio.so'
/tmp/.mount_audacifqpqC8/bin/audacity: /tmp/.mount_audacifqpqC8/lib/libselinux.so.1: no version information available (required by /lib/x86_64-linux-gnu/libgio-2.0.so.0)

(process:23084): Gdk-CRITICAL **: 11:13:39.388: IA__gdk_screen_get_root_window: assertion 'GDK_IS_SCREEN (screen)' failed

(process:23084): Gdk-CRITICAL **: 11:13:39.388: _gdk_pixmap_new: assertion '(drawable != NULL) || (depth != -1)' failed
../source_subfolder/src/gtk/bitmap.cpp(827): assert ""IsOk()"" failed in ConvertToImage(): invalid bitmap
../source_subfolder/src/common/image.cpp(2172): assert ""IsOk()"" failed in SetMaskColour(): invalid image
../source_subfolder/src/common/image.cpp(2223): assert ""IsOk()"" failed in SetMask(): invalid image
../source_subfolder/src/common/image.cpp(1853): assert ""IsOk()"" failed in GetWidth(): invalid image
../source_subfolder/src/common/image.cpp(1860): assert ""IsOk()"" failed in GetHeight(): invalid image
../source_subfolder/src/common/image.cpp(2112): assert ""IsOk()"" failed in GetAlpha(): invalid image
../source_subfolder/src/common/image.cpp(2232): assert ""IsOk()"" failed in HasMask(): invalid image
../source_subfolder/src/common/image.cpp(1986): assert ""IsOk()"" failed in GetData(): invalid image

(process:23084): GdkPixbuf-CRITICAL **: 11:13:39.395: gdk_pixbuf_new_from_data: assertion 'data != NULL' failed

(process:23084): Gdk-CRITICAL **: 11:13:39.395: IA__gdk_screen_get_root_window: assertion 'GDK_IS_SCREEN (screen)' failed

(process:23084): Gdk-CRITICAL **: 11:13:39.395: IA__gdk_drawable_get_display: assertion 'GDK_IS_DRAWABLE (drawable)' failed

(process:23084): Gdk-CRITICAL **: 11:13:39.395: IA__gdk_cursor_new_from_pixbuf: assertion 'GDK_IS_DISPLAY (display)' failed

(process:23084): GLib-GObject-CRITICAL **: 11:13:39.395: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
```

### Steps to reproduce

1. Launch the 3.3.1 AppImage on Linux (only tested on Ubuntu-based distros).

### Expected behavior

Audacity should start.

### Actual behavior

Audacity doesn't start.

### Audacity Version

latest stable version (from audacityteam.org/download)

### Operating system

Linux

## My System

* **OS:** KDE Neon 5.27 User Edition (Plasma Desktop 5.27.4, KDE Frameworks 5.105.0, Qt 5.15.9)
* **Linux Kernel:** 5.19.0-1023-lowlatency (64-bit)
* **Motherboard:** ASRock X58 Extreme3 (Intel X58 chipset)
* **CPU:** Intel Core i7-990x Gulftown (3.46 GHz hexa-core, Socket 1366) @ 4.15 GHz
* **RAM:** 24GB (6x G.Skill 4GB DDR3 PC3-10666 1333 MHz) @ 1430 MHz
* **Video:** EVGA NVIDIA GeForce GTX 980 Ti SC GAMING ACX 2.0+ w/ 6GB RAM
* **NVIDIA video driver:** 525.105.17

**Comment by mrbumpy409** [comment]
I just tested the previous 3.3.0 AppImage, and while it displays the same errors, it does actually start. So, I'm guessing the 3.3.1 AppImage failing to start is probably not related to the error messages.

**Comment by Ryan-Miller-Audacity** [comment]
I was able to reproduce this in my Linux VM.

**Comment by pablo-888** [comment]
> I have the same problem with the same symptoms/error messages on the Ubuntu-based Pop!_OS v22.04 operating system.
> 
> Can it be related to pipewire somehow, as I've set it as my playback device in Audacity's previous versions? I just don't know how to check that - by deactivating pipewire or switching to another playback device (e.g. ALSA) in Audacity.

I don't use pipewire and it still doesn't launch for me so I don't think it's that.

**Comment by MikePooh** [comment]
Linux Mint 21.1 Vera x64
Linux 5.15.0-71-generic x86_64
MATE 1.26.0
```
/lib/x86_64-linux-gnu/libatk-1.0.so.0
/lib/x86_64-linux-gnu/libatk-bridge-2.0.so.0
/lib/x86_64-linux-gnu/libcairo-gobject.so.2
/lib/x86_64-linux-gnu/libcairo.so.2
/lib/x86_64-linux-gnu/libgio-2.0.so.0
/lib/x86_64-linux-gnu/libglib-2.0.so.0
/lib/x86_64-linux-gnu/libgmodule-2.0.so.0
/lib/x86_64-linux-gnu/libgobject-2.0.so.0
/lib/x86_64-linux-gnu/libgthread-2.0.so.0
/lib/x86_64-linux-gnu/libjack.so.0
findlib: libportaudio.so: cannot open shared object file: No such file or directory
/home/mike/Applications/audacity-linux-3.3.1-x64.AppImage: Using fallback for library 'libportaudio.so'
/tmp/.mount_audaciKsca71/bin/audacity: /tmp/.mount_audaciKsca71/lib/libselinux.so.1: no version information available (required by /lib/x86_64-linux-gnu/libgio-2.0.so.0)
(process:5322): Gdk-CRITICAL **: 18:03:55.617: IA__gdk_screen_get_root_window: assertion 'GDK_IS_SCREEN (screen)' failed
(process:5322): Gdk-CRITICAL **: 18:03:55.617: _gdk_pixmap_new: assertion '(drawable != NULL) || (depth != -1)' failed
../source_subfolder/src/gtk/bitmap.cpp(827): assert ""IsOk()"" failed in ConvertToImage(): invalid bitmap
../source_subfolder/src/common/image.cpp(2172): assert ""IsOk()"" failed in SetMaskColour(): invalid image
../source_subfolder/src/common/image.cpp(2223): assert ""IsOk()"" failed in SetMask(): invalid image
../source_subfolder/src/common/image.cpp(1853): assert ""IsOk()"" failed in GetWidth(): invalid image
../source_subfolder/src/common/image.cpp(1860): assert ""IsOk()"" failed in GetHeight(): invalid image
../source_subfolder/src/common/image.cpp(2112): assert ""IsOk()"" failed in GetAlpha(): invalid image
../source_subfolder/src/common/image.cpp(2232): assert ""IsOk()"" failed in HasMask(): invalid image
../source_subfolder/src/common/image.cpp(1986): assert ""IsOk()"" failed in GetData(): invalid image
(process:5322): GdkPixbuf-CRITICAL **: 18:03:55.620: gdk_pixbuf_new_from_data: assertion 'data != NULL' failed
(process:5322): Gdk-CRITICAL **: 18:03:55.620: IA__gdk_screen_get_root_window: assertion 'GDK_IS_SCREEN (screen)' failed
(process:5322): Gdk-CRITICAL **: 18:03:55.620: IA__gdk_drawable_get_display: assertion 'GDK_IS_DRAWABLE (drawable)' failed
(process:5322): Gdk-CRITICAL **: 18:03:55.620: IA__gdk_cursor_new_from_pixbuf: assertion 'GDK_IS_DISPLAY (display)' failed
(process:5322): GLib-GObject-CRITICAL **: 18:03:55.620: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

```

**Comment by haleba-hotmail** [comment]
I'm having almost he exact same issue with Audacity 3.3.1 in kubuntu 18.04: it fails to load, but with complaints about not being able to find glibc versions 2.29 and 2.30:

    uname -srvp
    Linux 5.4.0-148-generic #165~18.04.1-Ubuntu SMP Thu Apr 20 01:14:06 UTC 2023 x86_64
    
    plasmashell --version
    plasmashell 5.12.9
    
    kf5-config --version
    Qt: 5.9.5
    KDE Frameworks: 5.44.0
    kf5-config: 1.0

        sudo lshw -short -c system -c bus -c memory -c processor
    H/W path            Device      Class          Description
    ==========================================================
                                    system         System Product Name (SKU)
    /0                              bus            ROG STRIX B450-I GAMING
    /0/0                            memory         64KiB BIOS
    /0/21                           memory         32GiB System Memory
    /0/21/0                         memory         16GiB DIMM DDR4 Synchronous Unbuffered (    Unregistered) 3200 MHz (0.3 ns)
    /0/21/1                         memory         16GiB DIMM DDR4 Synchronous Unbuffered (    Unregistered) 3200 MHz (0.3 ns)
    /0/23                           memory         384KiB L1 cache
    /0/24                           memory         2MiB L2 cache
    /0/25                           memory         4MiB L3 cache
    /0/26                           processor      AMD Ryzen 3 3200G with Radeon Vega Graphics

    ./audacity-linux-3.3.1-x64.appimage
    /usr/lib/x86_64-linux-gnu/libatk-1.0.so.0
    /usr/lib/x86_64-linux-gnu/libatk-bridge-2.0.so.0
    /usr/lib/x86_64-linux-gnu/libcairo-gobject.so.2
    /usr/lib/x86_64-linux-gnu/libcairo.so.2
    /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0
    /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0
    /usr/lib/x86_64-linux-gnu/libgmodule-2.0.so.0
    /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
    /usr/lib/x86_64-linux-gnu/libgthread-2.0.so.0
    /usr/lib/x86_64-linux-gnu/libjack.so.0
    findlib: libportaudio.so: cannot open shared object file: No such file or directory
    /home/barry-u/AppImages/audacity-linux-3.3.1-x64.appimage: Using fallback for library     'libportaudio.so'
    /tmp/.mount_audaciiPM06z/bin/audacity: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not     found (required by /tmp/.mount_audaciiPM06z/bin/audacity)
    /tmp/.mount_audaciiPM06z/bin/audacity: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not     found (required by /tmp/.mount_audaciiPM06z/lib/lib-project-file-io.so)
    /tmp/.mount_audaciiPM06z/bin/audacity: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not     found (required by /tmp/.mount_audaciiPM06z/lib/libgtk-x11-2.0.so.0)
    /tmp/.mount_audaciiPM06z/bin/audacity: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not     found (required by /tmp/.mount_audaciiPM06z/lib/lib-audio-io.so)
    /tmp/.mount_audaciiPM06z/bin/audacity: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not     found (required by /tmp/.mount_audaciiPM06z/lib/lib-track.so)
    /tmp/.mount_audaciiPM06z/bin/audacity: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not     found (required by /tmp/.mount_audaciiPM06z/lib/lib-numeric-formats.so)
    /tmp/.mount_audaciiPM06z/bin/audacity: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not     found (required by /tmp/.mount_audaciiPM06z/lib/lib-math.so)
    /tmp/.mount_audaciiPM06z/bin/audacity: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not     found (required by /tmp/.mount_audaciiPM06z/lib/libmpg123.so.0)
    /tmp/.mount_audaciiPM06z/bin/audacity: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not     found (required by /tmp/.mount_audaciiPM06z/lib/libsndfile.so.1)
    /tmp/.mount_audaciiPM06z/bin/audacity: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not     found (required by /tmp/.mount_audaciiPM06z/lib/libvorbis.so.0.4.9)
    /tmp/.mount_audaciiPM06z/bin/audacity: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not     found (required by /tmp/.mount_audaciiPM06z/lib/libFLAC.so.8)
    /tmp/.mount_audaciiPM06z/bin/audacity: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not     found (required by /tmp/.mount_audaciiPM06z/fallback/libportaudio.so/libportaudio.so)
    /tmp/.mount_audaciiPM06z/bin/audacity: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not     found (required by /tmp/.mount_audaciiPM06z/lib/libpng16.so.16)
    /tmp/.mount_audaciiPM06z/bin/audacity: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not     found (required by /tmp/.mount_audaciiPM06z/lib/libopus.so.0)
    /tmp/.mount_audaciiPM06z/bin/audacity: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not     found (required by /tmp/.mount_audaciiPM06z/lib/libpixman-1.so.0)
    /tmp/.mount_audaciiPM06z/bin/audacity: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.30' not     found (required by /tmp/.mount_audaciiPM06z/lib/libselinux.so.1)    


Audacity Appimage v. 3.1.3 that I downloaded from a source I can't remember also complains about libportaudio.so without any glibc errors, but loads just fine despite numerous Gdk errors

    ./audacity-linux-3.1.3-x86_64.appimage
    /usr/lib/x86_64-linux-gnu/libatk-1.0.so.0
    /usr/lib/x86_64-linux-gnu/libatk-bridge-2.0.so.0
    /usr/lib/x86_64-linux-gnu/libcairo-gobject.so.2
    /usr/lib/x86_64-linux-gnu/libcairo.so.2
    /usr/lib/x86_64-linux-gnu/libgmodule-2.0.so.0
    /usr/lib/x86_64-linux-gnu/libgthread-2.0.so.0
    /usr/lib/x86_64-linux-gnu/libjack.so.0
    findlib: libportaudio.so: cannot open shared object file: No such file or directory
    /home/barry-u/AppImages/audacity-linux-3.1.3-x86_64.appimage: Using fallback for library     'libportaudio.so'
    
    (process:4648): Gdk-CRITICAL **: 18:30:53.620: IA__gdk_screen_get_root_window: assertion     'GDK_IS_SCREEN (screen)' failed
    
    (process:4648): Gdk-CRITICAL **: 18:30:53.620: _gdk_pixmap_new: assertion '(drawable != NULL) || (    depth != -1)' failed
    /home/runner/work/audacity/audacity/conan-home/.conan/data/wxwidgets/3.1.3.1-audacity/_/_/build/    387c18a58e2fde1ef9c941329038b8fc62a56b98/source_subfolder/src/gtk/bitmap.cpp(827): assert     ""IsOk()"" failed in ConvertToImage(): invalid bitmap
    /home/runner/work/audacity/audacity/conan-home/.conan/data/wxwidgets/3.1.3.1-audacity/_/_/build/    387c18a58e2fde1ef9c941329038b8fc62a56b98/source_subfolder/src/common/image.cpp(2172): assert     ""IsOk()"" failed in SetMaskColour(): invalid image
    /home/runner/work/audacity/audacity/conan-home/.conan/data/wxwidgets/3.1.3.1-audacity/_/_/build/    387c18a58e2fde1ef9c941329038b8fc62a56b98/source_subfolder/src/common/image.cpp(2223): assert     ""IsOk()"" failed in SetMask(): invalid image
    /home/runner/work/audacity/audacity/conan-home/.conan/data/wxwidgets/3.1.3.1-audacity/_/_/build/    387c18a58e2fde1ef9c941329038b8fc62a56b98/source_subfolder/src/common/image.cpp(1853): assert     ""IsOk()"" failed in GetWidth(): invalid image
    /home/runner/work/audacity/audacity/conan-home/.conan/data/wxwidgets/3.1.3.1-audacity/_/_/build/    387c18a58e2fde1ef9c941329038b8fc62a56b98/source_subfolder/src/common/image.cpp(1860): assert     ""IsOk()"" failed in GetHeight(): invalid image
    /home/runner/work/audacity/audacity/conan-home/.conan/data/wxwidgets/3.1.3.1-audacity/_/_/build/    387c18a58e2fde1ef9c941329038b8fc62a56b98/source_subfolder/src/common/image.cpp(2112): assert     ""IsOk()"" failed in GetAlpha(): invalid image
    /home/runner/work/audacity/audacity/conan-home/.conan/data/wxwidgets/3.1.3.1-audacity/_/_/build/    387c18a58e2fde1ef9c941329038b8fc62a56b98/source_subfolder/src/common/image.cpp(2232): assert     ""IsOk()"" failed in HasMask(): invalid image
    /home/runner/work/audacity/audacity/conan-home/.conan/data/wxwidgets/3.1.3.1-audacity/_/_/build/    387c18a58e2fde1ef9c941329038b8fc62a56b98/source_subfolder/src/common/image.cpp(1986): assert     ""IsOk()"" failed in GetData(): invalid image
    
    (process:4648): GdkPixbuf-CRITICAL **: 18:30:53.622: gdk_pixbuf_new_from_data: assertion 'data !=     NULL' failed
    
    (process:4648): Gdk-CRITICAL **: 18:30:53.622: IA__gdk_screen_get_root_window: assertion     'GDK_IS_SCREEN (screen)' failed
    
    (process:4648): Gdk-CRITICAL **: 18:30:53.622: IA__gdk_drawable_get_display: assertion     'GDK_IS_DRAWABLE (drawable)' failed
    
    (process:4648): Gdk-CRITICAL **: 18:30:53.622: IA__gdk_cursor_new_from_pixbuf: assertion     'GDK_IS_DISPLAY (display)' failed
    
    (process:4648): GLib-GObject-CRITICAL **: 18:30:53.622: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

**Comment by razvaneduard** [comment]
kubuntu 22.04.2 LTS

 ./audacity-linux-3.3.1-x64.AppImage 
/lib/x86_64-linux-gnu/libatk-1.0.so.0
/lib/x86_64-linux-gnu/libatk-bridge-2.0.so.0
/lib/x86_64-linux-gnu/libcairo-gobject.so.2
/lib/x86_64-linux-gnu/libcairo.so.2
/lib/x86_64-linux-gnu/libgio-2.0.so.0
/lib/x86_64-linux-gnu/libglib-2.0.so.0
/lib/x86_64-linux-gnu/libgmodule-2.0.so.0
/lib/x86_64-linux-gnu/libgobject-2.0.so.0
/lib/x86_64-linux-gnu/libgthread-2.0.so.0
/lib/x86_64-linux-gnu/libjack.so.0
findlib: libportaudio.so: cannot open shared object file: No such file or directory
/home/raz/Downloads/audacity-linux-3.3.1-x64.AppImage: Using fallback for library 'libportaudio.so'
/tmp/.mount_audacib7Spih/bin/audacity: /tmp/.mount_audacib7Spih/lib/libselinux.so.1: no version information available (required by /lib/x86_64-linux-gnu/libgio-2.0.so.0)

(process:15871): Gdk-CRITICAL **: 20:54:14.200: IA__gdk_screen_get_root_window: assertion 'GDK_IS_SCREEN (screen)' failed

(process:15871): Gdk-CRITICAL **: 20:54:14.200: _gdk_pixmap_new: assertion '(drawable != NULL) || (depth != -1)' failed
../source_subfolder/src/gtk/bitmap.cpp(827): assert ""IsOk()"" failed in ConvertToImage(): invalid bitmap
../source_subfolder/src/common/image.cpp(2172): assert ""IsOk()"" failed in SetMaskColour(): invalid image
../source_subfolder/src/common/image.cpp(2223): assert ""IsOk()"" failed in SetMask(): invalid image
../source_subfolder/src/common/image.cpp(1853): assert ""IsOk()"" failed in GetWidth(): invalid image
../source_subfolder/src/common/image.cpp(1860): assert ""IsOk()"" failed in GetHeight(): invalid image
../source_subfolder/src/common/image.cpp(2112): assert ""IsOk()"" failed in GetAlpha(): invalid image
../source_subfolder/src/common/image.cpp(2232): assert ""IsOk()"" failed in HasMask(): invalid image
../source_subfolder/src/common/image.cpp(1986): assert ""IsOk()"" failed in GetData(): invalid image

(process:15871): GdkPixbuf-CRITICAL **: 20:54:14.203: gdk_pixbuf_new_from_data: assertion 'data != NULL' failed
(process:15871): Gdk-CRITICAL **: 20:54:14.203: IA__gdk_screen_get_root_window: assertion 'GDK_IS_SCREEN (screen)' failed
(process:15871): Gdk-CRITICAL **: 20:54:14.203: IA__gdk_drawable_get_display: assertion 'GDK_IS_DRAWABLE (drawable)' failed
(process:15871): Gdk-CRITICAL **: 20:54:14.203: IA__gdk_cursor_new_from_pixbuf: assertion 'GDK_IS_DISPLAY (display)' failed
(process:15871): GLib-GObject-CRITICAL **: 20:54:14.203: g_object_unref: assertion 'G_IS_OBJECT (object)' failed




exactly the same output is printed by 3.3.0 if I run it from console, and it starts ok

**Comment by Ryan-Miller-Audacity** [comment]
Testing this in my Ubuntu 22.04 VirtualBox VM, the appimage now runs as expected with the build from PR #4643 for 3.3.2. Upon opening other features within the app, they are functional as well.

**Comment by mpldr** [comment]
Hi, I am still experiencing this issue with the AppImage provided in the release and the Arch Package (which should be built using the default steps).

Possibly important detail: The pre-launch plugin download worked fine.

<details>
<summary>The log (pretty much the same as above</summary>
<pre><code>
$ ~/Downloads/audacity-linux-3.3.2-x64.AppImage
/usr/lib/libatk-1.0.so.0
/usr/lib/libatk-bridge-2.0.so.0
/usr/lib/libcairo-gobject.so.2
/usr/lib/libcairo.so.2
/usr/lib/libgio-2.0.so.0
/usr/lib/libglib-2.0.so.0
/usr/lib/libgmodule-2.0.so.0
/usr/lib/libgobject-2.0.so.0
/usr/lib/libgthread-2.0.so.0
/usr/lib/libjack.so.0
/usr/lib/libportaudio.so

(process:433389): Gdk-CRITICAL **: 19:30:08.104: IA__gdk_screen_get_root_window: assertion 'GDK_IS_SCREEN (screen)' failed

(process:433389): Gdk-CRITICAL **: 19:30:08.105: _gdk_pixmap_new: assertion '(drawable != NULL) || (depth != -1)' failed
../source_subfolder/src/gtk/bitmap.cpp(827): assert ""IsOk()"" failed in ConvertToImage(): invalid bitmap
../source_subfolder/src/common/image.cpp(2172): assert ""IsOk()"" failed in SetMaskColour(): invalid image
../source_subfolder/src/common/image.cpp(2223): assert ""IsOk()"" failed in SetMask(): invalid image
../source_subfolder/src/common/image.cpp(1853): assert ""IsOk()"" failed in GetWidth(): invalid image
../source_subfolder/src/common/image.cpp(1860): assert ""IsOk()"" failed in GetHeight(): invalid image
../source_subfolder/src/common/image.cpp(2112): assert ""IsOk()"" failed in GetAlpha(): invalid image
../source_subfolder/src/common/image.cpp(2232): assert ""IsOk()"" failed in HasMask(): invalid image
../source_subfolder/src/common/image.cpp(1986): assert ""IsOk()"" failed in GetData(): invalid image

(process:433389): GdkPixbuf-CRITICAL **: 19:30:08.108: gdk_pixbuf_new_from_data: assertion 'data != NULL' failed

(process:433389): Gdk-CRITICAL **: 19:30:08.108: IA__gdk_screen_get_root_window: assertion 'GDK_IS_SCREEN (screen)' failed

(process:433389): Gdk-CRITICAL **: 19:30:08.108: IA__gdk_drawable_get_display: assertion 'GDK_IS_DRAWABLE (drawable)' failed

(process:433389): Gdk-CRITICAL **: 19:30:08.108: IA__gdk_cursor_new_from_pixbuf: assertion 'GDK_IS_DISPLAY (display)' failed

(process:433389): GLib-GObject-CRITICAL **: 19:30:08.108: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
</pre></code></details>

And probably more useful in debugging: my environment. (I have also run it with `GDK_BACKEND` set to `x11` and `wayland`

```
SHELL=/bin/zsh
CSF_MDTVTexturesDirectory=/usr/share/opencascade/resources/Textures
CSF_DrawPluginDefaults=/usr/share/opencascade/resources/DrawResources
I3SOCK=/run/user/1000/sway-ipc.1000.1862.sock
LESS_TERMCAP_se=[0m
LESS_TERMCAP_so=[45;93m
CSF_LANGUAGE=us
CSF_MIGRATION_TYPES=/usr/share/opencascade/resources/StdResource/MigrationSheet.txt
CSF_OCCTResourcePath=/usr/share/opencascade/resources
XCURSOR_SIZE=14
CSF_STEPDefaults=/usr/share/opencascade/resources/XSTEPResource
GPG_TTY=/dev/pts/8
EDITOR=nvim
XDG_SEAT=seat0
PWD=/home/moritz/Games
LOGNAME=moritz
XDG_SESSION_TYPE=wayland
DRAWHOME=/usr/share/opencascade/resources/DrawResources
ECORE_EVAS_ENGINE=wayland_egl
SYSTEMD_EXEC_PID=1341
_=/usr/bin/env
CSF_StandardLiteDefaults=/usr/share/opencascade/resources/StdResource
ELM_DISPLAY=wl
MOTD_SHOWN=pam
HOME=/home/moritz
LANG=en_GB.UTF-8
_JAVA_AWT_WM_NONREPARENTING=1
XDG_CURRENT_DESKTOP=sway
GOSUMDB=off
SWAYSOCK=/run/user/1000/sway-ipc.1000.1862.sock
WAYLAND_DISPLAY=wayland-1
INVOCATION_ID=63b4b89ad5ff42e6b7305d02bb71bd7d
DOCKER_BUILDKIT=1
CSF_ShadersDirectory=/usr/share/opencascade/resources/Shaders
CSF_EXCEPTION_PROMPT=1
CSF_XmlOcafResource=/usr/share/opencascade/resources/XmlOcafResource
QT_QPA_PLATFORM=wayland-egl
CSF_SHMessage=/usr/share/opencascade/resources/SHMessage
MAKEFLAGS=-j 16
XDG_SESSION_CLASS=user
TERM=foot
LESS_TERMCAP_me=[0m
LESS_TERMCAP_md=[01;31m
USER=moritz
SDL_VIDEODRIVER=wayland
CSF_StandardDefaults=/usr/share/opencascade/resources/StdResource
CSF_IGESDefaults=/usr/share/opencascade/resources/XSTEPResource
VISUAL=nvim
GOPROXY=direct
DISPLAY=:0
CSF_XCAFDefaults=/usr/share/opencascade/resources/StdResource
LESS_TERMCAP_ue=[0m
SHLVL=1
MOZ_ENABLE_WAYLAND=1
LESS_TERMCAP_us=[01;32m
LC_MESSAGES=C
XDG_VTNR=1
CSF_PluginDefaults=/usr/share/opencascade/resources/StdResource
CSF_TObjMessage=/usr/share/opencascade/resources/TObj
XDG_SESSION_ID=1
CASROOT=/usr
XDG_RUNTIME_DIR=/run/user/1000
DEBUGINFOD_URLS=https://debuginfod.archlinux.org 
CSF_XSMessage=/usr/share/opencascade/resources/XSMessage
XCURSOR_THEME=Bibata-Original-Classic
MMGT_CLEAR=1
PATH=/home/moritz/.local/share/zinit/polaris/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/home/moritz/go/bin:/home/moritz/.npm-packages/bin:/home/moritz/.npm-packages/bin:/home/moritz/.bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/usr/lib/rustup/bin:/home/moritz/go/bin:/home/moritz/.npm-packages/bin:/home/moritz/.npm-packages/bin:/home/moritz/.bin:/home/moritz/.local/share/zinit/plugins/zpm-zsh---colorize/bin
CSF_TObjDefaults=/usr/share/opencascade/resources/StdResource
DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus
ELM_ENGINE=wayland_egl
HG=/usr/bin/hg
MAIL=/var/spool/mail/moritz
DRAWDEFAULT=/usr/share/opencascade/resources/DrawResources/DrawDefault
OLDPWD=/home/moritz/Games
ELM_ACCEL=opengl
TERM_PROGRAM=foot
TERM_PROGRAM_VERSION=1.14.0
COLORTERM=truecolor
ZPFX=/home/moritz/.local/share/zinit/polaris
ZSH_CACHE_DIR=/home/moritz/.cache/zinit
PMSPEC=0uUpiPsf
LESS= -R -M
SSH_AUTH_SOCK=/run/user/1000/ssh-agent.sock
SSH_AGENT_PID=2002
```

**Comment by erroreyes** [comment]
Sorry to add to this closed issue, but I just stumbled upon this too with v3.3.3. That said, I know that v3.3.2 and v3.3.1 did work fine, but now that I try to run them to compare with the latest version, those two also fail to start. I'm thinking that something changed in my system that is now preventing these 3 versions of Audacity from starting.

The logs form running the app image from the terminal looks just like the others posted above. The only extra bit that I can share is this:

```
Jun 26 17:46:03 rb-linux systemd[1385]: Started app-\x2fhome\x2froby\x2fApplications\x2faudacity\x2dlinux\x2d3.3.3\x2dx64-7d751c371bb94627ad5e6877333809ee.scope - /home/roby/Applications/audacity>
Jun 26 17:46:03 rb-linux systemd[1]: tmp-.mount_audaciJzVPbg.mount: Deactivated successfully.
Jun 26 17:46:10 rb-linux kwin_x11[1669]: qt.qpa.xcb: QXcbConnection: XCB error: 3 (BadWindow), sequence: 53239, resource id: 75497916, major code: 15 (QueryTree), minor code: 0
Jun 26 17:46:10 rb-linux systemd[1]: tmp-.mount_audacigFfkkv.mount: Deactivated successfully.
Jun 26 17:46:22 rb-linux systemd[1]: tmp-.mount_audacicg1RXa.mount: Deactivated successfully.
Jun 26 17:46:22 rb-linux kwin_x11[1669]: qt.qpa.xcb: QXcbConnection: XCB error: 3 (BadWindow), sequence: 54982, resource id: 75497916, major code: 15 (QueryTree), minor code: 0
Jun 26 17:46:34 rb-linux kernel: traps: audacity[237682] trap int3 ip:7ff61efc7c72 sp:7fff54795140 error:0 in lib-crashpad-configurer.so[7ff61efb1000+1b000]
Jun 26 17:46:40 rb-linux systemd[1]: tmp-.mount_audaciuHA7cc.mount: Deactivated successfully.
Jun 26 17:47:37 rb-linux plasmashell[1711]: libkcups: Renew-Subscription last error: 0 successful-ok
```

**Comment by 0pcom** [comment]
Apologies for posting on a closed issue, I did not see that anyone has been able to fix it thusfar.

I have this issue on archlinux with a fresh install of the latest version

```
$ sudo pacman -Qe audacity
audacity 1:3.4.2-1
```

There are several errors having to do with missing `.so` files.

After checking, it seems all of the supposedly missing `.so` files were provided by the package, but not installed to a path where they could be found / used

```
audacity: error while loading shared libraries: lib-theme-resources.so: cannot open shared object file: No such file or directory

$ sudo pacman -Fy lib-theme-resources.so
extra/audacity 1:3.4.2-1 (pro-audio) [installed]
    usr/lib/audacity/lib-theme-resources.so
```

I used the following command to symlink those `.so` files to `/usr/lib/`

```
sudo ln -sf /usr/lib/audacity/*.so /usr/lib/
```

After that, audacity starts without errors.

Perhaps it's not the right way to fix it but it does work, and it should fix it for any linux, until the packages are fixed.

When the package is fixed, it will be necessary to remove the conflicting symlinks.
